connector/connect/docs/client-connection-string.md

```diff
@@ -91,6 +91,16 @@ sc://hostname:port/;param1=value;param2=value
     <i>Default: </i><pre>_SPARK_CONNECT_PYTHON</pre> in the Python client</td>
     <td><pre>user_agent=my_data_query_app</pre></td>
   </tr>
+  <tr>
+    <td>session_id</td>
+    <td>String</td>
+    <td>In addition to the user ID, the cache of Spark Sessions in the Spark Connect
+    server uses a session ID as the cache key. This option in the connection string
+    allows to provide this session ID to allow sharing Spark Sessions for the same users
+    for example across multiple languages.<br/>
+    <i>Default: A UUID generated randomly</td>
```

Comment:
document that this must be an UUID string of the string of the format 00112233-4455-6677-8899-aabbccddeeff

Reply:
Out of curiosity: Do we depend on this format somewhere downstream?

Reply:
The error will throw an INVALID_HANDLE error on the server if it's not an UUID.
We do not have a hard dependency at this point, but processing it as UUID makes it possible to make assumptions in the future, e.g. that we can persist them as UUID, store them as BINARY(16) etc. If we don't lock it down, then we can't do this in the future because of possible old clients sending us arbitraty session_ids.
If we want to relax it in the future, we can - remove the check on the server, allow new clients to set other session ids.